File: Castro-Lara-Daniel_Ivan/Procesos/files C/fork.c
/*
    Sistemas Operativos
    Daniel Iván Castro Lara
    Programa en C que duplica el proceso actual
*/

#include <sys/types.h>
#include <stdio.h>
#include <unistd.h>
#include <time.h>
#include <stdlib.h>

int main() {
    pid_t pid;
    
        pid = fork();
        switch(pid){
            case -1: 
                printf("Error de creación\n");
            case 0: 
                printf("Código del hijo\n");
            default: 
                printf("Código del padre\n");
        }
}   

/*
    1.- gcc -o <ejecutable> <archivo_código>
    2.- <ejecutable>
*/

File: Castro-Lara-Daniel_Ivan/Procesos/files Java/fork.java
/*
    Sistemas Operativos
    Daniel Iván Castro Lara
    Programa en lenguaje Java
*/

public class ForkSimulation {
    public static void main(String[] args) {
        // Crear un nuevo hilo para simular el proceso hijo
        Thread childThread = new Thread(() -> {
            System.out.println("Código del hijo");
        });

        try {
            // Iniciar el hilo (simulando el fork del hijo)
            childThread.start();

            // Esperar a que el hilo hijo termine antes de continuar
            childThread.join();

            // Simulando el código del padre
            System.out.println("Código del padre");
        } catch (Exception e) {
            System.out.println("Error de creación: " + e.getMessage());
        }
    }
}

/* 
$ hilo_v1.java
*/
